itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -811,4 +819,120 @@ public void testIfCkptSetForObjectsByBootstrapReplLoad() throws Throwable {
+            .run("use " + replDbFromReplica)
+            .run("repl status " + replDbFromReplica)
+            .verifyResult(tupleReplica.lastReplicationId)
+            .run("show tables")
+            .verifyResults(new String[] { "t1" })
+            .run("select country from t1")
+            .verifyResults(Arrays.asList("india"));
+
+    // Check if DB/table/partition in C doesn't have repl.source.for props. Also ensure, ckpt property
+    // is set to bootstrap dump location used in C.
+    Database db = replica.getDatabase(replDbFromReplica);
+    verifyIfSrcOfReplPropMissing(db.getParameters());
+    verifyIfCkptSet(db.getParameters(), tupleReplica.dumpLocation);
+    Table t1 = replica.getTable(replDbFromReplica, "t1");
+    verifyIfCkptSet(t1.getParameters(), tupleReplica.dumpLocation);
+    Partition india = replica.getPartition(replDbFromReplica, "t1", Collections.singletonList("india"));
+    verifyIfCkptSet(india.getParameters(), tupleReplica.dumpLocation);
+
+    // Perform alters in A for incremental replication
+    WarehouseInstance.Tuple tuplePrimaryInc = primary.run("use " + primaryDbName)
+            .run("alter database " + primaryDbName + " set dbproperties('dummy_key'='dummy_val')")
+            .run("alter table t1 set tblproperties('dummy_key'='dummy_val')")
+            .run("alter table t1 partition(country='india') set fileformat orc")
+            .dump(primaryDbName, tuplePrimary.lastReplicationId);
+
+    // Incremental Repl A -> B with alters on db/table/partition
+    WarehouseInstance.Tuple tupleReplicaInc = replica.load(replicatedDbName, tuplePrimaryInc.dumpLocation)
+            .run("repl status " + replicatedDbName)
+            .verifyResult(tuplePrimaryInc.lastReplicationId)
+            .dump(replicatedDbName, tupleReplica.lastReplicationId);
+
+    // Check if DB in B have ckpt property is set to bootstrap dump location used in B and missing for table/partition.
+    db = replica.getDatabase(replicatedDbName);
+    verifyIfCkptSet(db.getParameters(), tuplePrimary.dumpLocation);
+    t1 = replica.getTable(replicatedDbName, "t1");
+    verifyIfCkptPropMissing(t1.getParameters());
+    india = replica.getPartition(replicatedDbName, "t1", Collections.singletonList("india"));
+    verifyIfCkptPropMissing(india.getParameters());
+
+    // Incremental Repl B -> C with alters on db/table/partition
+    replica.load(replDbFromReplica, tupleReplicaInc.dumpLocation)
+            .run("use " + replDbFromReplica)
+            .run("repl status " + replDbFromReplica)
+            .verifyResult(tupleReplicaInc.lastReplicationId);
+
+    // Check if DB/table/partition in C doesn't have repl.source.for props. Also ensure, ckpt property
+    // in DB is set to bootstrap dump location used in C but for table/partition, it is missing.
+    db = replica.getDatabase(replDbFromReplica);
+    verifyIfCkptSet(db.getParameters(), tupleReplica.dumpLocation);
+    verifyIfSrcOfReplPropMissing(db.getParameters());
```

Comment:
i think there is a inconsistency in the behavior ..for database the ckpt property is not removed after incremental ..but for table and partition it is removed.

Reply:
That's due to different behaviour for AlterDatabase and AlterTable in base code. However, this is not an issue as it doesn't affect state of target DB.